libkbfs/keybase_daemon_rpc.go

```diff
@@ -305,3 +306,46 @@ func (k *KeybaseDaemonRPC) Shutdown() {
 		k.shutdownFn()
 	}
 }
+
+// GetTLFCryptKeys implements the TlfKeysInterface interface for
+// KeybaseDaemonRPC.
+func (k *KeybaseDaemonRPC) GetTLFCryptKeys(
+	ctx context.Context, tlfName string) (res keybase1.TLFCryptKeys, err error) {
+	var tlfHandle *TlfHandle
+
+getHandle:
+	for {
+		tlfHandle, err = ParseTlfHandle(ctx, k.config.KBPKI(), tlfName, false)
+		switch e := err.(type) {
+		case TlfNameNotCanonical:
+			tlfName = e.NameToTry
+		case nil:
+			break getHandle
+		default:
+			return res, err
+		}
+	}
+
+	res.CanonicalName = keybase1.CanonicalTlfName(tlfHandle.GetCanonicalName())
+
+	var rmd ImmutableRootMetadata
+	var id TlfID
+	_, rmd, id, err = k.config.KBFSOps().GetOrInitializeNewMDMaster(
+		ctx, k.config.MDOps(), tlfHandle)
+	if err != nil {
+		return res, err
+	}
+	res.TlfID = keybase1.TLFID(id.String())
```

Comment:
Actually on second thought maybe getting it by name is better, since it means the caller can run Identify calls to make sure the participants on the folder are the same users they are trying to chat with.

But then I'm still wondering why the caller needs to know the ID.